### PR TITLE
model: LateOn-Code models definition

### PR DESCRIPTION
**Pull request by NohTow**
Hey,
This PR adds the LateOn-Code family models (both base and edge, as well as pre-trained and fine-tuned models)
cc @Samoed, let's chat about what to add/fix

Close https://github.com/embeddings-benchmark/mteb/issues/4094

### CODE REVIEW

File: mteb/models/model_implementations/pylate_models.py [hunk 1]
```diff
@@ -491,13 +489,189 @@ def _encode(
...
+        "php-Code",
+    ],
+    open_weights=True,
+    revision="71251a6ee61eee488de7e3ae29f5fb4c3c94699b",
+    public_training_code="https://github.com/lightonai/pylate/blob/main/examples/train/lateon_code/pre-training.py",
+    public_training_data="https://huggingface.co/datasets/lightonai/cornstack",
+    release_date="2026-02-12",
+    n_parameters=int(149 * 1e6),
+    n_embedding_parameters=38684160,
+    memory_usage_mb=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/pylate_models.py]: Can you fill this param? You can call `meta.calculate_memory_usage_mb(fetch_from_hf=True)`

File: mteb/models/model_implementations/pylate_models.py [hunk 2]
```diff
@@ -491,13 +489,189 @@ def _encode(
...
+        "go-Code",
+        "java-Code",
+        "javascript-Code",
+        "ruby-Code",
+        "php-Code",
+    ],
+    open_weights=True,
+    revision="71251a6ee61eee488de7e3ae29f5fb4c3c94699b",
+    public_training_code="https://github.com/lightonai/pylate/blob/main/examples/train/lateon_code/pre-training.py",
+    public_training_data="https://huggingface.co/datasets/lightonai/cornstack",
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/pylate_models.py:530]: <3

**Review by KennethEnevoldsen — APPROVED** [review]
Not much to add here - looks great! (seems like linting fails - you can fix this using `make lint` ([docs](https://embeddings-benchmark.github.io/mteb/CONTRIBUTING/#running-linting)).

File: mteb/models/model_implementations/pylate_models.py (hunk 2 above)
Comment (NohTow) [reply mteb/models/model_implementations/pylate_models.py:530]: I'll always make all of our releases as open as I possibly can! (appreciate the support in this matter, thanks! <3)

File: mteb/models/model_implementations/pylate_models.py (hunk 1 above)
Comment (NohTow) [reply mteb/models/model_implementations/pylate_models.py]: I filled it!
I also added embed_dim, not based on the mteb calculator but considering the dim outputed by our last dense layer (what we embed), let me know if it's ok
